File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SquareOfSums(count int) int {
	// sum of 1 to count
	sum := (1 + count) * count / 2
	return sum * sum
}

func SumOfSquares(count int) int {
	// sum of squares
	return count * (count + 1) * (2*count + 1) / 6
}

func Difference(count int) int {
	return SquareOfSums(count) - SumOfSquares(count)
}

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  VERSION = 2
  def initialize(n)
    @n = n
  end

  def square_of_sum
    ((1 + @n) * @n / 2)**2
  end

  def sum_of_squares
    @n * (@n + 1) * (2 * @n + 1) / 6
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
